Training/Training/Training.c

```diff
@@ -1,14 +1,34 @@
-// ------------------------------------------------------------------------------------------------
-// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
-// Copyright (c) Metamation India.
-// ------------------------------------------------------------------
-// Program.c
-// Program on main branch.
-// ------------------------------------------------------------------------------------------------
-
-#include<stdio.h>
-
-int main () {
-	printf ("Hello World!");
-	return 0;
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// training.c
+// Selection sort and binary search functions are defined
+// Sudarson S
+// ------------------------------------------------------------------------------------------------
+#include"training.h"
+
+int BinarySearch (int* arr, int arrSize, int key) {
+   int low = 0, high = arrSize - 1, idx = -1;
+   while (low <= high) {
+      int mid = low + (high - low) / 2;
+      // Checks if key is present at mid and is it's first occurrence
+      arr[mid] == key ? idx = mid, high = mid - 1 : arr[mid] < key ? low = mid + 1 : (high = mid - 1);

```

Comment:
Mostly ternary operator evaluates a Boolean expression and returns either one expression. Here it is mere assignment. So, 'if else' can be used.